### PR TITLE
update example of AST for TokenTypes.LAMBDA #9458

### LINKED ISSUE #9458: update example of AST for TokenTypes.LAMBDA
https://checkstyle.org/apidocs/com/puppycrawl/tools/checkstyle/api/TokenTypes.html#LAMBDA

needs to be extended to show example of AST.
Source to be updated: https://github.com/checkstyle/checkstyle/blob/master/src/main/java/com/puppycrawl/tools/checkstyle/api/TokenTypes.java

PLEASE READ for more details: https://github.com/checkstyle/checkstyle/wiki/Good-practice-on-how-to-update-AST-in-TokenTypes

### PR DESCRIPTION
**Pull request by Hrushi20**
Closes #9458

Screenshot (6)

Expected update example of AST for TokenTypes.LAMBDA

Screenshot (4)

Terminal view of AST

Screenshot (5)

```
public class Test{
    public static void main(String[] args) {
        int [] a = {1,2,3};

        Arrays.stream(a).map(el -> el);
    }
}

$ java -jar target/checkstyle-8.45-SNAPSHOT-all.jar -t ./src/main/java/Test.java

IMPORT -> import [1:0]
|--DOT -> . [1:16]
|   |--DOT -> . [1:11]
|   |   |--IDENT -> java [1:7]
|   |   `--IDENT -> util [1:12]
|   `--IDENT -> Arrays [1:17]
`--SEMI -> ; [1:23]
CLASS_DEF -> CLASS_DEF [3:0]
|--MODIFIERS -> MODIFIERS [3:0]
|   `--LITERAL_PUBLIC -> public [3:0]
|--LITERAL_CLASS -> class [3:7]
|--IDENT -> Test [3:13]
`--OBJBLOCK -> OBJBLOCK [3:17]
    |--LCURLY -> { [3:17]
    |--METHOD_DEF -> METHOD_DEF [4:4]
    |   |--MODIFIERS -> MODIFIERS [4:4]
    |   |   |--LITERAL_PUBLIC -> public [4:4]
    |   |   `--LITERAL_STATIC -> static [4:11]
    |   |--TYPE -> TYPE [4:18]
    |   |   `--LITERAL_VOID -> void [4:18]
    |   |--IDENT -> main [4:23]
    |   |--LPAREN -> ( [4:27]
    |   |--PARAMETERS -> PARAMETERS [4:28]
    |   |   `--PARAMETER_DEF -> PARAMETER_DEF [4:28]
    |   |       |--MODIFIERS -> MODIFIERS [4:28]
    |   |       |--TYPE -> TYPE [4:28]
    |   |       |   |--IDENT -> String [4:28]
    |   |       |   `--ARRAY_DECLARATOR -> [ [4:34]
    |   |       |       `--RBRACK -> ] [4:35]
    |   |       `--IDENT -> args [4:37]
    |   |--RPAREN -> ) [4:41]
    |   `--SLIST -> { [4:43]
    |       |--VARIABLE_DEF -> VARIABLE_DEF [5:7]
    |       |   |--MODIFIERS -> MODIFIERS [5:7]
    |       |   |--TYPE -> TYPE [5:7]
    |       |   |   |--LITERAL_INT -> int [5:7]
    |       |   |   `--ARRAY_DECLARATOR -> [ [5:10]
    |       |   |       `--RBRACK -> ] [5:11]
    |       |   |--IDENT -> a [5:13]
    |       |   `--ASSIGN -> = [5:15]
    |       |       `--ARRAY_INIT -> { [5:17]
    |       |           |--EXPR -> EXPR [5:18]
    |       |           |   `--NUM_INT -> 1 [5:18]
    |       |           |--COMMA -> , [5:19]
    |       |           |--EXPR -> EXPR [5:20]
    |       |           |   `--NUM_INT -> 2 [5:20]
    |       |           |--COMMA -> , [5:21]
    |       |           |--EXPR -> EXPR [5:22]
    |       |           |   `--NUM_INT -> 3 [5:22]
    |       |           |--COMMA -> , [5:23]
    |       |           |--EXPR -> EXPR [5:24]
    |       |           |   `--NUM_INT -> 4 [5:24]
    |       |           |--COMMA -> , [5:25]
    |       |           |--EXPR -> EXPR [5:26]
    |       |           |   `--NUM_INT -> 5 [5:26]
    |       |           `--RCURLY -> } [5:27]
    |       |--SEMI -> ; [5:28]
    |       |--EXPR -> EXPR [7:28]
    |       |   `--METHOD_CALL -> ( [7:28]
    |       |       |--DOT -> . [7:24]
    |       |       |   |--METHOD_CALL -> ( [7:21]
    |       |       |   |   |--DOT -> . [7:14]
    |       |       |   |   |   |--IDENT -> Arrays [7:8]
    |       |       |   |   |   `--IDENT -> stream [7:15]
    |       |       |   |   |--ELIST -> ELIST [7:22]
    |       |       |   |   |   `--EXPR -> EXPR [7:22]
    |       |       |   |   |       `--IDENT -> a [7:22]
    |       |       |   |   `--RPAREN -> ) [7:23]
    |       |       |   `--IDENT -> map [7:25]
    |       |       |--ELIST -> ELIST [7:32]
    |       |       |   `--LAMBDA -> -> [7:32]
    |       |       |       |--IDENT -> el [7:29]
    |       |       |       `--EXPR -> EXPR [7:35]
    |       |       |           `--IDENT -> el [7:35]
    |       |       `--RPAREN -> ) [7:37]
    |       |--SEMI -> ; [7:38]
    |       `--RCURLY -> } [9:4]
    `--RCURLY -> } [10:0]

```

### CODE REVIEW

File: src/main/java/com/puppycrawl/tools/checkstyle/api/TokenTypes.java [hunk 1]
```diff
@@ -4954,7 +4955,46 @@
     public static final int GENERIC_END = GeneratedJavaTokenTypes.GENERIC_END;
 
     /**
-     * Special lambda symbol {@code ->}.
+     * Lambda function "(() -&gt; {})"
+     *
+     * <p>For example: </p>
+     * <pre>
+     *  Arrays.stream(a).map(el -&gt; el);
```
Comment (pbludov) [inline src/main/java/com/puppycrawl/tools/checkstyle/api/TokenTypes.java:4962]: The code snippet should be compilable. For example, something like
`IntStream.of(1).forEach(n -> {});`

File: src/main/java/com/puppycrawl/tools/checkstyle/api/TokenTypes.java [hunk 2]
```diff
@@ -21,6 +21,8 @@
 
 import com.puppycrawl.tools.checkstyle.grammar.GeneratedJavaTokenTypes;
 
+import java.util.Arrays;
```
Comment (pbludov) [inline src/main/java/com/puppycrawl/tools/checkstyle/api/TokenTypes.java:24]: What the reason for this change? This import is not used.

**Review by pbludov — CHANGES_REQUESTED** [review]

File: src/main/java/com/puppycrawl/tools/checkstyle/api/TokenTypes.java [hunk 3]
```diff
@@ -4954,7 +4955,46 @@
...
+     * <p>For example: </p>
+     * <pre>
+     *  Arrays.stream(a).map(el -&gt; el);
+     * </pre>
+     * <p>Here a is an Array</p>
+     * <p>parses as:</p>
+     * <pre>
+     *    +--EXPR -&gt; EXPR
+     *                 |
+     *               +`--METHOD_CALL -&gt; (
```
Comment (romani) [inline src/main/java/com/puppycrawl/tools/checkstyle/api/TokenTypes.java:4969]: content here should exactly as from CLI, please use it.

**Review by romani — CHANGES_REQUESTED** [review]
item:

**Comment by romani** [comment]
Fixed in another PR

**Pull request closed without merging**